app/controllers/items_controller.rb

```diff
@@ -585,38 +659,64 @@ def apply_apo_defaults
   # add a workflow to an object if the workflow is not present in the active table
   def add_workflow
     return unless params[:wf]
-    wf = @object.workflows[params[:wf]]
-    # check for this workflow is present and active (not archived)
+    wf_name = params[:wf]
+    wf = @object.workflows[wf_name]
+    # check the workflow is present and active (not archived)
     if wf && wf.active?
-      render :status => 500, :text => "#{params[:wf]} already exists!"
+      render :status => 500, :text => "#{wf_name} already exists!"
       return
     end
-    @object.create_workflow(params[:wf])
+    @object.create_workflow(wf_name)
 
     # We need to sync up the workflows datastream with workflow service (using #find)
     # and then force a committed Solr update before redirection.
-    reindex Dor::Item.find(params[:id])
+    reindex find_druid(params[:id])
     Dor::SearchService.solr.commit
 
+    msg = "Added #{wf_name}"
     if params[:bulk]
-      render :text => "Added #{params[:wf]}"
+      render :text => msg
     else
-      redirect_to catalog_path(params[:id]), :notice => "Added #{params[:wf]}"
+      redirect_to catalog_path(params[:id]), notice: msg
     end
   end
 
   private
```

Comment:
Only reordered some methods into alphabetical order and changed before filter permission method names to the positive sense (like forbid_modify to allow_modify).